### ISSUE TITLE
snippets are not getting created

### ISSUE THREAD

**Issue by NavjotSingh007** [VincentKos/vscode-snippet-creator#1]
**vs code version**
Version: 1.30.2 (system setup)
Commit: 61122f88f0bf01e2ac16bdb9e1bc4571755f5bd8
Date: 2019-01-07T22:54:13.295Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
OS: Windows_NT x64 6.1.7601

**Comment by lmeurs** [comment]
I second that:

Version: 1.30.1
Commit: dea8705087adb1b5e5ae1d9123278e178656186a
Date: 2018-12-18T18:07:32.870Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
OS: Linux x64 4.15.0-39-generic

**Comment by VincentKos** [comment]
Can you check in the directory `%appdata%\Code\User\snippets` if any files is present ?

**Comment by NavjotSingh007** [comment]
I think your path is wrong. There is no Code folder just after AppData. In my Windows7, Actual Path is:
C:\Users\%username%\AppData\Roaming\Code\User\snippets

**Comment by VincentKos** [comment]
%appdata% is not the app data folder but the Roaming one :)

**Comment by VincentKos** [comment]
That's super weird. I can't really do anything. Are you sure you enter every prompt ? They are all mandatory and shouldn't left blank.

**Comment by NavjotSingh007** [comment]
Im sure I enter every prompt

**Comment by RobDesideri** [comment]
I have the same problem. After debugging the extension I found that (at least in my case) that the problem depends on whether the snippets are saved as JSON but in fact also contain **comments** (by default VSCode inserts comments into the file).

I already create a PR solving the problem. View #2 for details.

**Comment by NavjotSingh007** [comment]
ok, I got the error. whenever your extension finds any comments in a language.json file, Extension is not able to parse that json & Hence not able to write new data to that json

**Comment by VincentKos** [comment]
When #3 will be fixed i'll update the extension on vsce and see if it fix this.

**Comment by lankymart** [comment]
Also have this issue but believe is because I'm using VSCode Insiders which has a different path to the snippets `\Code - Insiders` instead of `\Code`, is that not a more intuitive way to determine the path?

**Comment by ldiebold** [comment]
Note this problem also occurs when there's trailing commas in the .json file.

**Comment by Pixney-William** [comment]
I have this issue, but only with Javascript files. And it has started to occur recently.

**Comment by Pixney-William** [comment]
Seems to be comments that is my problem in the javascript file as well.

**Comment by abulka** [comment]
Confirming what @lankymart discovered. Specifically when using the code insiders edition, and e.g. adding snippets for _restructuredtext_, the extension source code in 
`/Users/Andy/.vscode-insiders/extensions/vincentkos.snippet-creator-0.0.1/src/SnippetsManager.js`
puts snippets in

```
/Users/Andy/Library/Application\ Support/Code/User/snippets/restructuredtext.json
```
but that should be
```
/Users/Andy/Library/Application\ Support/Code\ -\ Insiders/User/snippets/restructuredtext.json 
```

in other words

```
Code
```
needs to be replaced with
```
Code\ -\ Insiders
```

**Comment by abulka** [comment]
Confirming what @Pixney-William discovered. Whilst the built in *vscode* snippet system can miraculously read from a .json snippets file containing comments within the JSON, this extention cannot. Technically JSON does not allow comments.

Ironically the default snippet file created by vscode *does* contain a block of comments, so if you leave that in your snippet file (as many users would), this extension will fail to add to it.

The line that fails in this extension is:
```javascript
let snippets = JSON.parse(text.toString());
```
which fails silently if there are comments in the json file. 

Perhaps the extension needs to preparse the json file and strip any comments? But it might be considered rude to arbitrarily delete them - how to preserve the comments in the json?  

Perhaps the answer is to add a `try... catch` with a failure warning the user: "the JSON snippet file XXXX failed to parse - possibly due to comments - please remove any comments before using this extension again" or something.

**Comment by abulka** [comment]
I've added a pull request #9 which addresses most of the issues above, incl. being compatible with the code-insiders edition.

- Detects and also works with VSCode code-insiders edition
- Repairs blank prefix and blank description if user doesn't enter them
- Attempting to add a snippet with an existing name now succeeds (a unique id is added to the incoming snippet)
- Better success announcement message
- increments to 0.0.6 so that it supercedes the deprecated 0.0.5 version more obviously
- A bit more helpful info added to the readme

Interestingly the version I checked out from master *does not fail* when comments are in the JSON snippets file, because the extension **already uses** the more resilient https://www.npmjs.com/package/jsonc-parser JSON parser.  Not sure why my marketplace installed version of this extension did not have these parsing improvements - perhaps the version in master was not version incremented nor published to the marketplace?

If anybody wants to try my changes to extension, download the new [.vsix file](http://bit.ly/snippet-creator-0-6) and install manually via the vscode extensions menu "Install from VSIX..."
